### PR TITLE
Update German translation

### PR DESCRIPTION
**Pull request by lucaszischka**
Hello there 👋,

Im currently updating the German translation, because since it was first translated a lot in the original readme changed. For example im going to add the contributors list, because it is currently missing. Also I will go through the text and update all typos, grammatical errors or unclear sections.

The only thing im going to change, is that I will add my name to the contributors list. If that's not allowed, because im a first time contributor and im only translating, I will remove it of course.

### CODE REVIEW

**Comment by lucaszischka** [comment]
@iluwatar Im not sure why the sonar pr job fails, as there are no code changes. However, im ready for a review now.

**Comment by iluwatar** [comment]
The build failure is unrelated and has already been fixed in the master branch

File: README.md [hunk 1]
```diff
@@ -343,6 +343,7 @@ This project is licensed under the terms of the MIT license.
...
       <td align="center"><a href="https://www.linkedin.com/in/karshilsheth/"><img src="https://avatars.githubusercontent.com/u/37497287?v=4?s=100" width="100px;" alt="Karshil sheth"/><br /><sub><b>Karshil sheth</b></sub></a><br /><a href="https://github.com/iluwatar/java-design-patterns/commits?author=karshil2309" title="Code">💻</a></td>
       <td align="center"><a href="https://github.com/kongleong86"><img src="https://avatars.githubusercontent.com/u/3745455?v=4?s=100" width="100px;" alt="kongleong86"/><br /><sub><b>kongleong86</b></sub></a><br /><a href="https://github.com/iluwatar/java-design-patterns/commits?author=kongleong86" title="Code">💻</a></td>
+      <td align="center"><a href="https://github.com/lucaszischka"><img src="https://avatars.githubusercontent.com/u/63545066?v=4?s=100" width="100px;" alt="Lucas Zischka"/><br /><sub><b>Lucas Zischka</b></sub></a><br /><a href="https://github.com/iluwatar/java-design-patterns/commits?author=lucaszischka" title="Translation">🌍</a></td>
```
Comment (iluwatar) [inline README.md:346]: No need to add this manually. We will add your name here once the pull request is merged.

Comment (iluwatar) [reply README.md:346]: Please revert this edit and we can merge the changes.

**Review by iluwatar — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (lucaszischka) [reply README.md]: Done :)

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Thank you for the translation update!

@all-contributors please add @lucaszischka for docs

**Comment by iluwatar** [comment]
https://github.com/all-contributors please add @lucaszischka for translation

**Pull request merged**